src/constructs/autoscaling/user-data.test.ts

```diff
@@ -76,30 +130,69 @@ describe("GuUserData", () => {
         bucketName: "test-app-config",
         files: ["secrets.json", "application.conf"],
       },
+      additionalStatements: [
+        GuUserData.stripMargin`
+        |mkdir /etc/gu
+        |cat > /etc/gu/my-application.conf <<-'EOF'
+        |  include "application"
+        |EOF`,
+      ],
     };
 
-    const { userData } = new GuUserData(stack, props);
+    createAsg(stack, props);
 
-    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
-      vpc,
-      userData,
-      stageDependentProps: {
-        [Stage.CODE]: {
-          minimumInstances: 1,
-        },
-        [Stage.PROD]: {
-          minimumInstances: 3,
+    expect(stack).toHaveResource("AWS::AutoScaling::LaunchConfiguration", {
+      UserData: {
+        "Fn::Base64": {
+          "Fn::Join": [
+            "",
+            [
+              "#!/bin/bash\nmkdir -p $(dirname '/etc/testing/secrets.json')\naws s3 cp 's3://test-app-config/secrets.json' '/etc/testing/secrets.json'\nmkdir -p $(dirname '/etc/testing/application.conf')\naws s3 cp 's3://test-app-config/application.conf' '/etc/testing/application.conf'\n\nmkdir /etc/gu\ncat > /etc/gu/my-application.conf <<-'EOF'\n  include \"application\"\nEOF\nmkdir -p $(dirname '/testing/my-app.deb')\naws s3 cp 's3://",
+              {
+                Ref: "DistributionBucketName",
+              },
+              "/test-stack/",
+              {
+                Ref: "Stage",
+              },
+              "/testing/my-app.deb' '/testing/my-app.deb'\ndpkg -i /testing/my-app.deb",
+            ],
+          ],
         },
       },
     });
+  });
+
+  test("additionalStatements should run between configuration download and distributable download", () => {
+    const stack = simpleGuStackForTesting();
+
+    const props: GuUserDataProps = {
+      distributable: {
+        bucket: new GuDistributionBucketParameter(stack),
+        fileName: "my-app.deb",
+        executionStatement: `dpkg -i /${stack.app}/my-app.deb`,
+      },
+      configuration: {
+        bucketName: "test-app-config",
+        files: ["secrets.json", "application.conf"],
+      },
+      additionalStatements: [
+        "mkdir /etc/gu",
+        "cat > /etc/gu/my-application.conf <<-'EOF'",
+        `  include "application"`,
+        "EOF",
+      ],
```

Comment:
`GuUserData.stripMargin` is a utility function that returns a `string` rather than a custom type. I think your suggestion would result in `additionalStatements` being `string | string[]` which might be confusing?